actions/analyze_rosbag_image_exposure/src/analyze_rosbag_image_exposure/__main__.py

```diff
@@ -0,0 +1,485 @@
+            else:
+                put_tag = False
+                good_images.append(img_path)
+            
+            if put_tag:
+                bag_put_tags.append(put_tag)
+                if dataset and extract_all_images:
+                    print(f"putting tag {put_tag} on imag {img_path}")
+                    FileManager.put_tags(f"{bag_folder.name}/{img_path}", [put_tag])
+
+
+        # put tags on rosbag if dataset is not None and put_tags exist
+        if dataset: 
+            if bag_put_tags:
+                #bag_fname = pathlib.Path(input_file_or_folder) / f"{bag_folder.name}.bag"
+                bag_fname = f"{bag_folder.name}.bag"
+                # build a MetadataChangeset with put_tag direction
+                changeset = updates.MetadataChangeset().Builder()
+                # put tags on the Changeset builder one at a time
+                for tag in set(bag_put_tags): # only tag dataset with each tag value once
+                    changeset = changeset.put_tag(tag)
+                changeset = changeset.build()
+                bag_file = dataset.get_file_info(bag_fname)
+                file_update_request = files.file_requests.UpdateFileRecordRequest(metadata_changeset = changeset)
+                # pass request to add metadata
+                bag_file = bag_file.update(file_update_request)
+                logger.info(f"Tagging {bag_fname} with {bag_put_tags}")
+
+        # save some example images, max 5 for each of over and underexposed
+        max_overex_examples = min(len(overexposed_images), 5)
+        overexposed_examples = overexposed_images[:max_overex_examples]
+
+        if overexposed_examples:
+            overexposed_outputs = output_root / f"{bag_folder.name}_overexposed_examples"
+            overexposed_outputs.mkdir(parents=True,exist_ok=True)
+
+        # copy overexposed example images to output example dir
+            for img_path in overexposed_examples:
+                shutil.copy(bag_folder / img_path, overexposed_outputs / pathlib.Path(img_path).name)
+        
+
+        max_underex_examples = min(len(underexposed_images), 5)
+        underexposed_examples = underexposed_images[:max_underex_examples]
+        if underexposed_examples:
+            underexposed_outputs = output_root / f"{bag_folder.name}_underexposed_examples"
+            underexposed_outputs.mkdir(parents=True, exist_ok=True)
+
+            # copy overexposed example images to output example dir
+            for img_path in underexposed_examples:
+                shutil.copy(bag_folder / img_path, underexposed_outputs / pathlib.Path(img_path).name)
+        
+        if not extract_all_images:
+            # remove output bag of images that aren't examples
+            shutil.rmtree(bag_folder)
+        else:
+            # apply updates from File Metadata Manager to dataset
+            if dataset:
+                FileManager.apply_to_dataset(dataset)
+
+
+def process_and_maybe_save_video(
```

Comment:
let's get rid of this function (we don't need videos) and only use process_rosbag